File: app/src/main/java/com/jorgedguezm/elections/PartyColorBinder.java
package com.jorgedguezm.elections;

import android.graphics.Color;
import android.view.View;
import android.widget.SimpleAdapter;
import android.widget.TextView;

/**
 * Created by jorge on 24/11/16.
 */

public class PartyColorBinder implements SimpleAdapter.ViewBinder {
    @Override
    public boolean setViewValue(View view, Object data, String textRepresentation) {
        if (view instanceof TextView && data.toString().matches("[#].*")) {
            view.setBackgroundColor(Color.parseColor(data.toString()));
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/jorgedguezm/elections/view/binders/PartyColorBinder.kt
package com.jorgedguezm.elections.view.binders

import android.graphics.Color
import android.view.View
import android.widget.SimpleAdapter
import android.widget.TextView

class PartyColorBinder: SimpleAdapter.ViewBinder {

    override fun setViewValue(view: View?, data: Any?, textRepresentation: String?): Boolean {
        if (view is TextView && data.toString().matches(Regex("[#].*"))) {
            view.setBackgroundColor(Color.parseColor(data.toString()))
            return true
        }

        return false
    }
}